submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/helpers/SubmissionFilesSource.kt

```diff
@@ -0,0 +1,44 @@
+package ac.uk.ebi.biostd.submission.helpers
+
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionPersistenceQueryService
+import ebi.ac.uk.extended.model.ExtFile
+import ebi.ac.uk.extended.model.ExtSubmission
+import ebi.ac.uk.extended.model.FireFile
+import ebi.ac.uk.extended.model.StorageMode.FIRE
+import ebi.ac.uk.extended.model.StorageMode.NFS
+import ebi.ac.uk.io.sources.DbFile
+import ebi.ac.uk.io.sources.FilesSource
+import ebi.ac.uk.io.sources.PathSource
+import ebi.ac.uk.model.Attribute
+import uk.ac.ebi.fire.client.integration.web.FireClient
+import java.io.File
+
+class SubmissionFilesSource(
+    private val sub: ExtSubmission,
+    private val nfsFiles: PathSource,
+    private val fireClient: FireClient,
+    private val previousVersionFiles: Map<String, ExtFile>,
+    private val queryService: SubmissionPersistenceQueryService,
+) : FilesSource {
+    override val description: String
+        get() = "Files of submission '${sub.accNo}', version ${sub.version}"
```

Comment:
description is shared with user and they are not aware of this so I think we just simple said previos version.